pageserver/src/page_cache.rs

```diff
@@ -381,6 +400,36 @@ impl PageCache {
         }
     }
 
+    // Section 1.3: Public interface functions for working with immutable file pages.
+
+    pub fn read_immutable_buf(&self, file_id: u64, blkno: u32) -> ReadBufResult {
+        let mut cache_key = CacheKey::ImmutableFilePage { file_id, blkno };
+
+        self.lock_for_read(&mut cache_key)
+    }
+
+    /// Immediately drop all buffers belonging to given file, without writeback
+    pub fn drop_buffers_for_immutable(&self, drop_file_id: u64) {
+        for slot_idx in 0..self.slots.len() {
```

Comment:
Right now default page cache size is 8192. It seems to be too small. Size of page cache has critical impact on system performance, because page reconstruction is about 10x times slower than just reading page from image layer or page cache. Size of page cache should not be small than checkpoint_distance and taken in account that it is shared between multiple tenant,it may be reasonable to multiply checkpoint_distance N times. From the other side, our VM nodes have large enough amount of memory and it is quite reasonable to use some noticeable part of it for page cache (there are not so much other memory consumers at page server except file system cache).

So, the reasonable value for page cache size will be several gigabytes (millions pages). I afraid that in this case such iteration though all page case slots will be rather expensive.

Reply:
Agreed. We want to configure the page cache to use up most of the RAM on the page server host.

> So, the reasonable value for page cache size will be several gigabytes (millions pages). I afraid that in this case such iteration though all page case slots will be rather expensive.

Yeah, it's a known problem with PostgreSQL too, which uses a similar clock algorithm for the page cache. We don't drop buffers very often, though, so I don't think it will be a huge problem even if it's expensive. The page cache is a very self-contained component of the system, so I think we can leave it as it is for now, and address later if we start to see problems. It won't affect the design of the rest of the system.